app/grandchallenge/github/views.py

```diff
@@ -0,0 +1,55 @@
+import hashlib
+import hmac
+import json
+from datetime import timedelta
+from secrets import compare_digest
+
+from django.conf import settings
+from django.db.transaction import non_atomic_requests
+from django.http import HttpResponse, HttpResponseForbidden
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils import timezone
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+
+from grandchallenge.github.models import GitHubWebhookMessage
+
+
+@csrf_exempt
+@require_POST
+@non_atomic_requests
+def github_webhook(request):
+    signature = hmac.new(
+        bytes(settings.GITHUB_WEBHOOK_SECRET, encoding="utf8"),
+        msg=request.body,
+        digestmod=hashlib.sha256,
+    ).hexdigest()
+    signature = f"sha256={signature}"
+    if not compare_digest(
+        signature, request.headers.get("X-Hub-Signature-256", "")
+    ):
+        return HttpResponseForbidden(
+            "Signatures do not match", content_type="text/plain",
+        )
+
+    GitHubWebhookMessage.objects.filter(
+        created__lte=timezone.now() - timedelta(days=7)
+    ).delete()
```

Comment:
Why are all the old messages being deleted in the POST view? Won't this delete all of the tarballs too? If this is needed, I would have thought that it should be done in a periodic task.